### ISSUE TITLE
DevHome freezes when trying to preview/switch between CPU and GPU quickly after opening window

### ISSUE THREAD

**Issue by luislhg** [microsoft/devhome#1188]
### Dev Home version

0.200.170.0

### Windows build number

10.0.22621.1848

### Steps to reproduce the bug

1) Open Dev Home
2) +Add Widget
3) Click on GPU and then CPU quickly (these are just my first two options)
4) Software freezes forever, have to end task in task manager

### Expected result

I expected GPU and CPU to be displayed normally without freezing the application forever. Regardless if I just opened the window and tried to preview a different widget.

### Actual result

Reproducing the issue (mp4 video)
https://github.com/microsoft/devhome/assets/16278387/5bb6bec6-d812-4e17-bd71-f1155d50f3c3

Terminal printscreen, seems like it got stuck in a loop forever
![image](https://github.com/microsoft/devhome/assets/16278387/3f5deaae-0ac5-43dd-bfeb-09d412aab186)

**Comment by krschau** [comment]
I think we need to be smarter here about what we show in the dialog. These widgets are heavy. We probably don't need an actual live graph or statistics while still in the dialog. Maybe we show a screenshot. Some of these widgets do have settings (like which GPU you want info for) so maybe the right thing to do is show real data, but not have it update continuously until the widget is actually pinned.

**Comment by aqibbanday** [comment]
Your PC is probably slow. Its working fine on my PC.

**Comment by luislhg** [comment]
@aqibbanday I don't think this is hardware related. The issue happens consistently when you switch to another widget preview before the first one has finished updating for the first time (when you have just opened the Add Widget window).

The PC specs are: Core i9-9900K, 32GB RAM, SSD NVM.e, GTX 2070 Super.

**Comment by Strypper** [comment]
> @aqibbanday I don't think this is hardware related. The issue happens consistently when you switch to another widget preview before the first one has finished updating for the first time (when you have just opened the Add Widget window). After the first widget and preview is fetched correctly, I can freely switch between any widget preview and it always works from that point.
> 
> The PC specs are: Core i9-9900K, 32GB RAM, 1TB SSD NVM.e, RTX 2070 Super.

back fire 🔥🔥🔥 hard bro

**Comment by luislhg** [comment]
Just wanted to update that since the last updates (currently running 0.701) all the widgets are way faster and I could not reproduce this crash anymore.

**Comment by krschau** [comment]
@luislhg Thanks for pointing this out! We did make some changes to the code that executes when switching between widgets like that, so I'm glad it seems to have fixed this scenario.
